src/fund/policies/risk-management/PriceTolerance.sol

```diff
@@ -1,121 +1,106 @@
 // SPDX-License-Identifier: GPL-3.0
 pragma solidity 0.6.8;
 
-import "../TradingSignatures.sol";
-import "../../hub/Spoke.sol";
 import "../../../dependencies/DSMath.sol";
-import "../../../dependencies/token/IERC20.sol";
-import "../../../integrations/interfaces/IOasisDex.sol";
-import "../../../prices/primitives/IPriceSource.sol";
+import "../../../fund/shares/Shares.sol";
+import "../../../prices/ValueInterpreter.sol";
+import "../../../registry/Registry.sol";
+import "../utils/CallOnIntegrationPostValidatePolicyBase.sol";
 
 /// @title PriceTolerance Contract
 /// @author Melon Council DAO <security@meloncoucil.io>
 /// @notice Validate the price tolerance of a trade
-contract PriceTolerance is TradingSignatures, DSMath {
-    enum Applied { pre, post }
+contract PriceTolerance is DSMath, PolicyBase, CallOnIntegrationPostValidatePolicyBase {
+    event PriceToleranceSet(address policyManager, uint256 value);
 
-    uint256 public tolerance;
+    uint256 internal constant ONE_HUNDRED_PERCENT = 10 ** 18;  // 100%
 
-    uint256 constant MULTIPLIER = 10 ** 16; // to give effect of a percentage
-    uint256 constant DIVISOR = 10 ** 18;
+    mapping (address => uint256) public policyManagerToPriceTolerance;
 
-    // _tolerance: 10 equals to 10% of tolerance
-    constructor(uint256 _tolerancePercent) public {
-        require(_tolerancePercent <= 100, "Tolerance range is 0% - 100%");
-        tolerance = mul(_tolerancePercent, MULTIPLIER);
-    }
-
-    /// @notice Taken from OpenZeppelin (https://git.io/fhQqo)
-   function signedSafeSub(int256 _a, int256 _b) internal pure returns (int256) {
-        int256 c = _a - _b;
-        require((_b >= 0 && c <= _a) || (_b < 0 && c > _a));
+    constructor(address _registry) public PolicyBase(_registry) {}
 
-        return c;
-    }
+    // EXTERNAL FUNCTIONS
 
-    function checkPriceToleranceTakeOrder(
-        address _makerAsset,
-        address _takerAsset,
-        uint256 _fillMakerQuantity,
-        uint256 _fillTakerQuantity
-    )
-        internal
-        view
-        returns (bool)
-    {
-        IPriceSource pricefeed = IPriceSource(
-            IRegistry(IHub(Spoke(msg.sender).HUB()).REGISTRY()).priceSource()
+    /// @notice Add the initial policy settings for a fund
+    /// @param _encodedSettings Encoded settings to apply to a fund
+    /// @dev A fund's PolicyManager is always the sender
+    /// @dev Only called once, on PolicyManager.enablePolicies()
+    function addFundSettings(bytes calldata _encodedSettings) external override onlyPolicyManager {
+        uint256 priceTolerance = abi.decode(_encodedSettings, (uint256));
+        require(priceTolerance > 0, "addFundSettings: priceTolerance must be greater than 0");
```

Comment:
Tough, because then we can't validate the input, but after discussing with the others, I think you're right about allowing `0` as a valid amount and then just leaving it up to frontends to properly validate their inputs.

Reply:
Fixed and pushed